modules/fedimint-walletv2-common/src/config.rs

```diff
@@ -0,0 +1,261 @@
+    /// The mint module will charge a non-configurable base fee of one thousand
+    /// satoshis per transaction input and output to account for the costs
+    /// incurred by the federation for processing the transaction. On top of
+    /// that the federation may charge a additional relative fee per input and
+    /// output of up to ten thousands parts per million which equals one
+    /// percent.
+    ///
+    /// # Errors
+    /// - This constructor returns an error if the relative fee is in excess of
+    ///   one thousand parts per million.
+    pub fn new(parts_per_million: u64) -> anyhow::Result<Self> {
+        anyhow::ensure!(
+            parts_per_million <= 10_000,
+            "Relative fee over one thousand parts per million is excessive"
+        );
+
+        Ok(Self {
+            base: Amount::from_sats(1000),
+            parts_per_million,
+        })
+    }
+
+    pub fn fee(&self, amount: Amount) -> Amount {
+        Amount::from_msats(self.fee_msats(amount.msats))
+    }
+
+    fn fee_msats(&self, msats: u64) -> u64 {
+        msats
+            .saturating_mul(self.parts_per_million)
+            .saturating_div(1_000_000)
+            .checked_add(self.base.msats)
+            .expect("The division creates sufficient headroom to add the base fee")
+    }
+}
+
+#[test]
+fn test_fee_consensus() {
+    let fee_consensus = FeeConsensus::new(10_000).expect("Relative fee is within range");
+
+    assert_eq!(
+        fee_consensus.fee(Amount::from_msats(99)),
+        Amount::from_sats(1000)
+    );
+
+    assert_eq!(
+        fee_consensus.fee(Amount::from_sats(1)),
+        Amount::from_msats(10) + Amount::from_sats(1000)
+    );
+
+    assert_eq!(
+        fee_consensus.fee(Amount::from_sats(1000)),
+        Amount::from_sats(10) + Amount::from_sats(1000)
+    );
+
+    assert_eq!(
+        fee_consensus.fee(Amount::from_bitcoins(1)),
+        Amount::from_sats(1_000_000) + Amount::from_sats(1000)
+    );
+
+    assert_eq!(
+        fee_consensus.fee(Amount::from_bitcoins(10_000)),
+        Amount::from_bitcoins(100) + Amount::from_sats(1000)
+    );
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize, Encodable, Decodable)]
+pub struct WalletClientConfig {
+    /// The public keys for the bitcoin multisig
+    pub bitcoin_pks: BTreeMap<PeerId, PublicKey>,
+    /// Confirmations required for a bitcoin transaction to be confirmed
+    pub finality_delay: u64,
+    /// The minimum amount a user can send on chain
+    pub dust_limit: bitcoin::Amount,
+    /// Fees taken by the guardians to process wallet inputs and outputs
+    pub fee_consensus: FeeConsensus,
+    /// Bitcoin network (e.g. testnet, bitcoin)
+    pub network: Network,
+}
+
+impl std::fmt::Display for WalletClientConfig {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "WalletClientConfig {}",
+            serde_json::to_string(self).map_err(|_e| std::fmt::Error)?
```

Comment:
Do we really need `Display` on this one? The implementation here looks like of like `Debug` anyway, not very usable for ... "output to the end user", if you ask me, and probably the whole struct shouldn't need it really in the first place. `{:?}` in formatting will produce roughly same thing without `serde_json`.